sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,76 @@ export class EventHubClient {
    */
   constructor(
     host: string,
-    entityPath: string,
+    eventHubPath: string,
     credentials: ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials,
     options?: ClientOptions
   );
   constructor(
-    host: string,
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
-      );
-    }
+    let connectionString;
+    let sharedAccessKeyName = "defaultKeyName";
+    let sharedAccessKey = "defaultKeyValue";
+    let eventHubPath: string | undefined;
+    let tokenProvider = {} as TokenProvider;
 
-    if (!entityPath) {
-      throw new Error("Please provide a value for the 'entityPath' parameter.");
-    }
+    hostOrConnectionString = String(hostOrConnectionString);
+    if (typeof eventHubPathOrOptions === "string") {
+      eventHubPath = eventHubPathOrOptions;
+      if (!eventHubPath) {
+        throw new Error("Please provide a value for the 'entityPath' parameter.");
+      }
+      if (!tokenProviderOrCredentials || typeof {}) {
```

Comment:
What is the `typeof {}` for here?

Reply:
This is just a leftover from testing. I'll remove it.

I was just testing this scenario 
```
let tokenProvider = {} as SasTokenProvider;
const client = new EventHubClient("host", "eventHubPath", tokenProvider);
```
and was thinking, below code will give keyName/key is not a property of null error:
```
 if (tokenProvider instanceof SasTokenProvider) {
          sharedAccessKeyName = tokenProvider.keyName;
          sharedAccessKey = tokenProvider.key;
        }
```
but turns out if I pass tokenprovider like this `let tokenProvider = {} as TokenProvider;`, then it is a typeof "object" and it doesn't go inside the if block. In this case we are using defaults of sharedAccessKeyName and sharedAccessKey .

Reply:
`typeof` will not return the class name. It has the below limited return values
- number
- string
- boolean
- undefined
- object